src/contextimpl.rs

```diff
@@ -237,8 +237,12 @@ impl<A> ContextImpl<A> where A: Actor, A::Context: AsyncContext<A>
             while self.handles.len() > 2 {
                 let handle = self.handles.pop().unwrap();
                 for idx in 0..self.items.len() {
-                    if self.items[idx].0 == handle {
-                        self.items.swap_remove(idx);
+                    if idx == self.items.len() {
```

Comment:
i think better solution would be to replace `for` with `while`.

Reply:
I checked it again and I'm still not sure how handles are structures... 
Just to understand: is the behaviour intended to be like this:

```rust
            // check cancelled handles, just in case
            while self.handles.len() > 2 {
                if let Some(handle) = self.handles.pop() {
                    self.handles.retain(|h| h != &handle);
                }
            }
```

[SmallVec::retain](https://docs.rs/smallvec/0.6.0/src/smallvec/lib.rs.html#637-648) is slightly more expensive than the hand backed `swap_remove` solution, but way easier.